### PR TITLE
add pypi twine check (see #994)

### PR DESCRIPTION
**Pull request by svenfuchs**
Can be tested using:

```yaml
deploy:
  provider: pypi
  edge:
    branch: sf-twine-check
⋮
```

### CODE REVIEW

File: lib/dpl/providers/pypi.rb [hunk 1]
```diff
@@ -29,12 +29,14 @@ class Pypi < Provider
            upload_docs:  'Uploading documentation (skip using "skip_upload_docs: true")'
 
       cmds setup_py:     'python setup.py %{distributions}',
+           twine_check:  'twine check',
```
Comment (di) [inline lib/dpl/providers/pypi.rb]:
```suggestion
           twine_check:  'twine check dist/*',
```

Comment (svenfuchs) [reply lib/dpl/providers/pypi.rb]: thank you @di!

File: spec/dpl/providers/pypi_spec.rb [hunk 2]
```diff
@@ -49,6 +50,10 @@
     it { should have_run 'twine upload --skip-existing -r pypi dist/*' }
   end
 
+  describe 'given --no_twine_check' do
+    it { should have_run 'twine check' }
```
Comment (di) [inline spec/dpl/providers/pypi_spec.rb]:
```suggestion
    it { should have_run 'twine check dist/*' }
```

**Comment by svenfuchs** [comment]
@di awesome, thanks for the feedback!

i'll merge this once it's green.

**Pull request merged**